src/components/BalancePage/BalancedTable.tsx

```diff
@@ -58,11 +88,9 @@ const BalancedTable = () => {
                         <NavLink to={'/user/balance/details/' + i.toString()}>
                           <InfoIcon />
                         </NavLink>
-                        <NavLink to={'/user/balance/edit/' + i.toString()}>
-                          <EditIcon />
-                        </NavLink>
+                        <EditIcon onClick={() => handleEditGoal(goal)} />
                         <div>
-                          <DeleteIcon />
+                          <DeleteIcon onClick={() => handleDeleteGoal(goal)} />
```

Comment:
we are using `goals?.map(goal, i)` where i is an index, so we can pass it to edit & delete function with no need to calculate current index of goal